packages/internet_detector.vm/tools/internet_detector.pyw

```diff
@@ -0,0 +1,238 @@
+
+    def set_icon(self, icon):
+        self.icon = icon
+        self.nid = (self.hwnd, 0, win32gui.NIF_ICON | win32gui.NIF_MESSAGE | win32gui.NIF_TIP, 
+                    win32con.WM_USER + 20, self.icon, self.tooltip)
+        win32gui.Shell_NotifyIcon(win32gui.NIM_MODIFY, self.nid)
+
+    def __del__(self):
+        # Remove the icon from the system tray when the object is destroyed
+        win32gui.Shell_NotifyIcon(win32gui.NIM_DELETE, self.nid)
+
+# Attempt to extract a known good value in response.
+def extract_title(data):
+    match = re.search(r'<title>(.*?)</title>', data)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+def check_internet():
+    for url, expected_response in CONNECT_TEST_URL_AND_RESPONSES.items():
+        try:
+            # Perform internet connectivity tests
+            response = requests.get(url, timeout=5, verify=False)
+            if expected_response in (extract_title(response.text) or response.text):
+                print(f"Internet connectivity detected via URL: {url}")
+                return True
+        except:
+            pass
+    return False
+
+def check_internet_and_update_tray_icon():
+    global tray_icon, hicon_indicator_off, hicon_indicator_on
+    if check_internet():
+        tray_icon.set_icon(hicon_indicator_on)
+        tray_icon.set_tooltip("Internet Connection: Detected")
+        # Set the background to internet connection background
+        if get_wallpaper_path() != INTERNET_BACKGROUND:     # Checked so program isn't continuously setting the wallpaper
+            set_wallpaper(INTERNET_BACKGROUND)
+    else: 
+        tray_icon.set_icon(hicon_indicator_off)
+        tray_icon.set_tooltip("Internet Connection: Not Detected")
+        # Reset background when internet is not detected
+        if get_wallpaper_path() != DEFAULT_BACKGROUND:      # Checked so program isn't continuously setting the wallpaper
+            set_wallpaper(DEFAULT_BACKGROUND)
+
+def check_internet_loop():
+    while not stop_event.is_set():
+        check_internet_and_update_tray_icon()
+        time.sleep(CHECK_INTERVAL)
+
+def tray_icon_loop():
+    global hwnd, tray_icon, hicon_indicator_off, hicon_indicator_on, stop_event
+    # Load icons
+    hicon_indicator_off = load_icon(ICON_INDICATOR_OFF)
+    hicon_indicator_on = load_icon(ICON_INDICATOR_ON)
+
+    # Wait for hwnd to be initialized
+    while hwnd is None:
+        time.sleep(0.1)
+
+    if hicon_indicator_off is None or hicon_indicator_on is None:
+        print("Error: Failed to load icons. Exiting TrayIconThread.")
+        return
+    
+    tray_icon = SysTrayIcon(hwnd, hicon_indicator_off, "Internet Detector")
+    
+    while not stop_event.is_set():
+        msg = win32gui.PeekMessage(hwnd, 0, 0, 0)
+        if msg and len(msg) == 6:
+            win32gui.TranslateMessage(msg)
+            win32gui.DispatchMessage(msg)
+        time.sleep(0.1)
+
+def enable_transparency_effects():
```

Comment:
I think the transparency effect does not work in a dark taskbar in non-activated Windows (at least I see no difference). What about in addition **setting the taskbar to rose color in the dark mode**? 

I am ok with keeping the transparency if you like it (I will never see it anyway :rofl:).